test/integration/http-integration-test.js

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable max-nested-callbacks */
+'use strict';
+const assert = require('assert-diff');
+const _ = require('lodash');
+const jayson = require('jayson');
+
+const createHTTPServer = require('../../src/index').createHTTPServer;
+
+const apiFixtures = require('../fixtures');
+const apiRequests = apiFixtures.requests;
+const apiResponses = apiFixtures.responses;
+
+const fixtures = require('./fixtures');
+
+const TIMEOUT = 20000;   // how long before each test case times out
+
+const apiOptions = {
+  server: 'wss://s1.ripple.com'
+};
+
+const httpPort = 3000;
+
+function createClient() {
+  return jayson.client.http({port: httpPort, hostname: 'localhost'});
+}
+
+const address = 'r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59';
+
+function makePositionalParams(params) {
+  return params.map(value => value[_.keys(value)[0]]);
+}
+
+function makeNamedParams(params) {
+  return _.reduce(params, _.assign, {});
+}
+
+function random() {
+  return _.fill(Array(16), 0);
+}
+
+describe('http server integration tests', function() {
+  this.timeout(TIMEOUT);
+
+  let server = null;
+  let client = null;
+
+  function createTestInternal(testName, methodName, testFunc, id, params) {
+    it(testName, function() {
+      return new Promise((resolve, reject) => {
+        client.request(methodName, params, id, (err, result) => {
+          if (err) {
+            return reject(err);
+          }
+          testFunc(result);
+          resolve();
+        });
+      });
+    });
+  }
+
+  function createTest(name, testFunc, id, params) {
+    createTestInternal(name + ' - positional params', name,
+      testFunc, id, makePositionalParams(params));
+    createTestInternal(name + ' - named params', name,
+      testFunc, id, makeNamedParams(params));
+  }
+
+  beforeEach(function(done) {
+    server = createHTTPServer(apiOptions, httpPort);
+    server.start().then(() => {
+      this.client = createClient();
+      client = this.client;
+      done();
+    }).catch(done);
+  });
+
+  afterEach(function(done) {
+    server.stop().then(done, done);
+  });
+
+
+  createTest('getLedgerVersion', (result) => {
+    assert(_.isNumber(result.result));
+  }, '001', []);
```

Comment:
How about using a property on the test suite to store an incrementing counter for the request ID?